src/box/wal.c

```diff
@@ -63,6 +63,8 @@ const char *wal_mode_STRS[WAL_MODE_MAX] = {
 
 int wal_dir_lock = -1;
 
+struct rlist wal_on_batch_write = RLIST_HEAD_INITIALIZER(wal_on_batch_write);
```

Comment:
Nit: You could write:

```C
struct RLIST_HEAD(wal_on_batch_write);
```

Maybe, name it simply `wal_on_write`? (Looks like there's no such thing as non-batch write.)

Also, the declaration needs a comment that triggers are actually invoked in tx, upon confirming a write.

Reply:
<details><summary>Done:</summary>

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index a980a8e3d..8b8c7594a 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -1983,7 +1983,7 @@ box_wait_vclock(const struct vclock *vclock, double deadline)
                .waiter = fiber(),
        };
        trigger_create(&on_wal_write, box_wait_vclock_f, &data, NULL);
-       trigger_add(&wal_on_batch_write, &on_wal_write);
+       trigger_add(&wal_on_write, &on_wal_write);
        do {
                if (fiber_yield_deadline(deadline))
                        break;
diff --git a/src/box/wal.c b/src/box/wal.c
index aa9440682..59d249cd3 100644
--- a/src/box/wal.c
+++ b/src/box/wal.c
@@ -63,7 +63,7 @@ const char *wal_mode_STRS[WAL_MODE_MAX] = {
 
 int wal_dir_lock = -1;
 
-struct rlist wal_on_batch_write = RLIST_HEAD_INITIALIZER(wal_on_batch_write);
+RLIST_HEAD(wal_on_write);
 
 static int
 wal_write_async(struct journal *, struct journal_entry *);
@@ -376,7 +376,7 @@ tx_complete_batch(struct cmsg *msg)
        /* Update the tx vclock to the latest written by wal. */
        vclock_copy(&replicaset.vclock, &batch->vclock);
        tx_schedule_queue(&batch->commit);
-       trigger_run(&wal_on_batch_write, NULL);
+       trigger_run(&wal_on_write, NULL);
        mempool_free(&writer->msg_pool, container_of(msg, struct wal_msg, base));
 }
 
@@ -594,7 +594,7 @@ wal_free(void)
                panic_syserror("WAL writer: thread join failed");
        }
 
-       trigger_destroy(&wal_on_batch_write);
+       trigger_destroy(&wal_on_write);
 
        wal_writer_destroy(writer);
 }
diff --git a/src/box/wal.h b/src/box/wal.h
index 7185d56bc..8940416aa 100644
--- a/src/box/wal.h
+++ b/src/box/wal.h
@@ -77,9 +77,10 @@ extern const char *wal_mode_STRS[];
 extern int wal_dir_lock;
 
 /**
- * Triggers invoked after WAL write confirmation for a batch of transactions.
+ * Triggers ivoked by tx thread's sched fiber upon WAL write confirmation
+ * arrival.
  */
-extern struct rlist wal_on_batch_write;
+extern struct rlist wal_on_write;
 
 #if defined(__cplusplus)
 extern "C" {
```
</details>